src/leaflet_d3.js

```diff
@@ -23,32 +23,44 @@ L.CartoDBd3Layer = L.Class.extend({
 		var self = this;
 		options = options || {};
 		L.Util.setOptions(this, options);
+		this.providers = {
```

Comment:
I would create a index.js within providers *package* so you can `var providers = require('./providers');` and later you can do something like `new provider.Sql(...)`.